### DISCUSSION TITLE
tar_render() for files generated in earlier targets

### DISCUSSION THREAD

**Discussion by CharnelMouse** [ropensci/tarchetypes#100] [Help] [answered]
I currently have a targets workflow that has something like this:
```{r}
list(
  tar_target(dummy_Rmd, generate_rmd(), format = "file")
  tar_render(new_html, "rmd_file_path")
)
```
However, running `tar_make()` returns an error that the file doesn't exist, so `tar_render` doesn't notice that other targets generate the file. Is there a good way to do this?

The context is that I have an Rmd written by someone else, outside of the targets workflow, and my version of `generate_rmd()` creates a modified version of it that changes file reads in that Rmd (`read.csv` etc.) to use `tar_read` or `tar_load` instead.

**Comment by wlandau** [comment, marked as answer]
tar_render() tries to detect existing dependencies in existing reports, which affects the structure of the dependency graph in a way that needs to happen before the pipeline begins, so it’s not a great fit for an Rmd written inside a target. You could write the Rmd before the pipeline begins or above the target list in _targets.R.
